pages/manifesto.tsx

```diff
@@ -36,18 +62,45 @@ const Manifesto = () => {
   }, [])
 
   useEffect(() => {
-    ;(async () => {
+    const fetchAllSignatures = async () => {
       const data = await (await fetch("./api/get_signatures")).json()
-      const userSigned = data.filter(
-        (item) => address === item.fields["Address"]
-      )
-      setHasSigned(userSigned.length > 0)
-      if (userSigned.length) setSignature(userSigned[0].fields["Signature"])
-    })()
+      // Create a Set of unique addresses with their signatures and timestamps, keeping latest signature for duplicates
+      const signatures = {}
```

Comment:
Using an object as a hash-map here ensures each address is only used once, and includes that latest signature (latest message driven)